### ISSUE TITLE
devon formatter not having same values like Sonar plugin

### ISSUE THREAD

**Issue by rhobolic** [devonfw/sonar-devon4j-plugin#136]
If I'm using the devon ide (devon4j) without touching everything I get issues in Sonar though I use the devon sonar plugin also without changes.

Problem description:
As an example here is to take the checkstyle rule which is also part of devon java profile "Intendation". When I look into the profile I see the following configuration.
![image](https://user-images.githubusercontent.com/99953472/154644174-63b231b5-9df3-442c-9f10-2a025b69ee11.png)
But if I look into the formatter in the ide there is the following configuration
![image](https://user-images.githubusercontent.com/99953472/154644560-594191ef-bca1-49dc-a9f9-e44ca498cd1b.png)
Therefore Sonar is throwing an issue. There are also other issues like "Import order" or "Line length"

Expectation
Devon ide, devon formatter and devon sonar plugin have the same values and work out of the box without throwing issues.

**Comment by hohwille** [comment]
@rhobolic thanks for reporting this bug.
Indeed this is very odd and kind of frustrating.
I am not sure why this happened now but my assumption is that the defaults changed with a sonarqube or checkstyle update.
After all the default rules and especially the whitespace rules for checkstyle are defined here:
https://github.com/devonfw/sonar-devon4j-plugin/blob/3bd6482d1164e7d915ec4d10779eeda6235316fe/src/main/resources/com/devonfw/ide/sonarqube/common/rules/devon4j/devon4j.xml#L1082

As we can see these rules rely on the defaults what is never a good idea:
https://github.com/devonfw/sonar-devon4j-plugin/blob/3bd6482d1164e7d915ec4d10779eeda6235316fe/src/main/resources/com/devonfw/ide/sonarqube/common/rules/devon4j/devon4j.xml#L1085

I will simply change this and set the indentation to 2 instead of 4.

BTW:
> If I'm using the devon ide (devon4j) without touching everything I get issues in Sonar though I use the devon sonar plugin also without changes.

As a lot of people seemed to get confused with devonfw artifacts here are some hints to match them properly:
* devonfw-ide is the tool to setup your local development environment: https://github.com/devonfw/ide#devonfw-ide - it is very generic and can be used for any kind of project (using Java, or .NET, or python, or NodeJS, or ...).
* devon4j is the Java stack of devonfw: https://github.com/devonfw/devon4j#devon4j
* sonar-devon4j-plugin is this repo with the sonar plugin for devon4j that you found perfectly and therefore filed the issues in the correct spot.

**Comment by hohwille** [comment]
For the record: There are several very picky checkstyle rules active in our config that should also be discussed:
* JavadocTagContinuationIndentationCheck - is that really something to verify and address on the level of SonarQube. Eclipse might format this different from IntelliJ and in the end it does not matter at all for real quality so discussing about things like this is pure waste.
* CyclomaticComplexityCheck with a max of 10 might also be kind of dangerous. While in general this is not such a bad idea, the metric is defined in a very stupid way and some case statements or if cascades can easily cause a complexity higher than 10. If this results in developers splitting the code that belongs perfectly together into separate methods just at some random point only for the same of the rule the quality will not increase but decrease.

There might be more in this regard. I can do some testing on this and file a new issue for it if applicable...

**Comment by hohwille** [comment]
I assume that this issue is fixed and can be closed. However, my tests are currently blocked by #141.

**Comment by hohwille** [comment]
Another interesting fact is that the file that I initially created came from here and is still present:
https://github.com/devonfw/ide-settings/blob/master/sonarqube/profiles/Devon-Java.xml

It has correct indentation configured and lots of other differences.
So something went wrong when PR #56 was implemented.

**Comment by hohwille** [comment]
I again have to give up for today:
I step-wise out-commented rule by rule when it was reported as error on startup.
This process was so annoying as we have so many rules, that I automated it step by step.
After full automation all rules had been disabled.
So to give an obvious example:
```
Caused by: java.lang.IllegalStateException: Rule with key 'devon4j:N8' not found
```
So SonarQube can not find a rule we define by our own plugin.
To me this means that SonarQube release is broken or changed in an incompatible way.
Our entire quality profile generation stopped working (at least with SonarQube 9.4.0.54424).
